File: Memento/Go/memento/Memento.go
package memento

type Memento struct {
  temp_record string
  temp_time int
}

func NewMemento(temp_record string, temp_time int) *Memento {
  return &Memento{temp_record, temp_time}
}

func (this *Memento) GetTempRecord() string {
  return this.temp_record
}

func (this *Memento) GetTempTime() int {
  return this.temp_time
}

File: Memento/PHP/Memento.php
<?php
  class Memento {
    private $temp_record;

    private $temp_time;

    public function __construct($temp_record, $temp_time) {
      $this->temp_record = $temp_record;
      $this->temp_time = $temp_time;
    }

    public function getTempRecord() {
      return $this->temp_record;
    }

    public function getTempTime() {
      return $this->temp_time;
    }
  }
